util/lru/lru.go

```diff
@@ -68,6 +68,12 @@ func (c *Cache[K, V]) Set(key K, value V) {
 	}
 }
 
+// Clear removes all items from the cache.
+func (c *Cache[K, V]) Clear() {
+	c.head = nil
```

Comment:
how about `*c = Cache[K,V]{MaxEntries: c.MaxEntries}`

Reply:
Not a huge fan of the "assign to struct receiver" pattern, thing; it's a bit confusing if you haven't seen it.